python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -99,6 +100,38 @@ def __post_init__(self) -> None:
             raise TypeError("pass_read_limit must be an int")
 
 
+def default_blocksize(scheduler: str) -> int:
+    """Return the default blocksize."""
+    try:
+        # Use PyNVML to find the worker device size.
+        import pynvml
+
+        pynvml.nvmlInit()
+        index = os.environ.get("CUDA_VISIBLE_DEVICES", "0").split(",")[0]
+        if index and not index.isnumeric():  # pragma: no cover
+            # This means index is UUID. This works for both MIG and non-MIG device UUIDs.
+            handle = pynvml.nvmlDeviceGetHandleByUUID(str.encode(index))
+        else:
+            # This is a device index
+            handle = pynvml.nvmlDeviceGetHandleByIndex(int(index))
+        device_size = pynvml.nvmlDeviceGetMemoryInfo(handle).total
```

Comment:
I don't know if we necessarily care about MIG, however, if someone attempts to use this with MIG this code is wrong, which I realized was a bug in Dask-CUDA. We should use https://github.com/rapidsai/dask-cuda/pull/1476/files#diff-fae45034757fc51999601c0f9886684027d56d2db243beb98ac08ef380731da1R92-R121 directly from Dask-CUDA, or make a copy of that function to get the correct handle.

Reply:
We can't depend on dask here, so we will need to copy code, or just not support MIG (which may be fine, the user can always configure the blocksize if they get a bad default in an edge case).